packages/tree-widget/src/components/header/TreeHeader.tsx

```diff
@@ -13,50 +13,86 @@ export interface TreeHeaderComponentProps {
   showAll?: () => void;
   hideAll?: () => void;
   invert?: () => void;
+  toggle2D?: () => void;
+  toggle2DIcon?: string;
+  toggle3D?: () => void;
+  toggle3DIcon?: string;
   searchOptions: SearchOptions;
 }
 
-export function TreeHeaderComponent({ searchOptions, showAll, hideAll, invert }: TreeHeaderComponentProps) {
+export function TreeHeaderComponent({
+  searchOptions,
+  showAll,
+  hideAll,
+  invert,
+  toggle2D,
+  toggle2DIcon,
+  toggle3D,
+  toggle3DIcon,
+}: TreeHeaderComponentProps) {
   return (
-      <SearchBar
-        value=""
-        className="tree-widget-header-tree-search-bar"
-        valueChangedDelay={500}
-        placeholder={TreeWidget.translate("search")}
-        title={TreeWidget.translate("searchForSomething")}
-        filteringInProgress={searchOptions.isFiltering}
-        onFilterCancel={searchOptions.onFilterCancel}
-        onFilterClear={searchOptions.onFilterCancel}
-        onFilterStart={searchOptions.onFilterStart}
-        onSelectedChanged={searchOptions.onResultSelectedChanged}
-        resultCount={searchOptions.matchedResultCount ?? 0}
-      >
-        <div>
-          {showAll &&
+    <SearchBar
+      value=""
+      className="tree-widget-header-tree-search-bar"
+      valueChangedDelay={500}
+      placeholder={TreeWidget.translate("search")}
+      title={TreeWidget.translate("searchForSomething")}
+      filteringInProgress={searchOptions.isFiltering}
+      onFilterCancel={searchOptions.onFilterCancel}
+      onFilterClear={searchOptions.onFilterCancel}
+      onFilterStart={searchOptions.onFilterStart}
+      onSelectedChanged={searchOptions.onResultSelectedChanged}
+      resultCount={searchOptions.matchedResultCount ?? 0}
+    >
+      <div>
+        {showAll && (
           <IconButton
             className={"tree-widget-header-tree-toolbar-icon"}
             key="show-all-btn"
             icon="icon-visibility"
             title={TreeWidget.translate("showAll")}
             onClick={showAll}
-          />}
-          {hideAll &&
+          />
+        )}
+        {hideAll && (
           <IconButton
             className={"tree-widget-header-tree-toolbar-icon"}
             key="hide-all-btn"
             icon="icon-visibility-hide-2"
             title={TreeWidget.translate("hideAll")}
             onClick={hideAll}
-          />}
-          {invert &&
+          />
+        )}
+        {invert && (
           <IconButton
             key="invert-all-btn"
             className={"tree-widget-header-tree-toolbar-icon"}
             title={TreeWidget.translate("invert")}
             icon="icon-visibility-invert"
             onClick={invert}
-          />}
-        </div>
-      </SearchBar>
-  )
-}
\ No newline at end of file
+          />
+        )}
+        {toggle2D && toggle2DIcon && (
+          <IconButton
+            className={"tree-widget-models-tree-toolbar-icon"}
```

Comment:
will the 2d, and 3d toggle only be available for the models tree? right now this TreeHeader component will be used by all the trees. So using `models` in the classname here isnt great imo